validphys2/src/validphys/overfit_metric.py

```diff
@@ -0,0 +1,128 @@
+"""
+overfit_metric.py
+
+This calculates the overfit metric
+"""
+
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy.stats as stats
+
+from reportengine import collect
+from reportengine.figure import figure
+from reportengine.table import table
+
+log = logging.getLogger(__name__)
+
+preds = collect("predictions", ("pdfs","dataset_inputs",))
+
+
+def array_expected_delta_chi2(calculate_chi2s_per_replica, replica_data):
+
+    res_over = calculate_chi2s_per_replica
+
+    fitted_val_erf = np.array([info.validation for info in replica_data])
+
+    number_pdfs = res_over.shape[0]
+    list_expected_delta_chi2 = []
+    for _ in range(number_pdfs * 1000):
+        mask = np.random.randint(0, number_pdfs, size=int(0.95 * number_pdfs))
+        res_tmp = res_over[mask][:, mask]
+
+        fitted_val_erf_tmp = fitted_val_erf[mask]
+        expected_val_chi2 = res_tmp.mean(axis=0)
+        delta_chi2 = fitted_val_erf_tmp - expected_val_chi2
+        expected_delta_chi2 = delta_chi2.mean()
+
+        list_expected_delta_chi2.append(expected_delta_chi2)
+    return np.array(list_expected_delta_chi2)
+
+
+def _create_new_val_pseudodata(pdf_data_index, fit_data_indices_list):
+    vl_data_fitrep = []
+    for fitreplica_info in fit_data_indices_list:
+        vl_data_fitrep.append(fitreplica_info.pseudodata.loc[pdf_data_index.val_idx])
+    return np.array(vl_data_fitrep).squeeze()
```

Comment:
The `squeeze` without argument can be tricky because (4,2,1) will go to (4,2) but (4,1,1) will go to (4,). I think it is better to specify the axis (or axes) that you want to squeeze.